syft/cataloger/javascript/cataloger.go

```diff
@@ -7,12 +7,21 @@ import (
 	"github.com/anchore/syft/syft/cataloger/common"
 )
 
-// NewJavascriptCataloger returns a new JavaScript cataloger object.
-func NewJavascriptCataloger() *common.GenericCataloger {
+// NewJavascriptCataloger returns a new JavaScript cataloger object based on detection of npm based packages.
+func NewJavascriptPackageCataloger() *common.GenericCataloger {
```

Comment:
The doc comment should match the function name ("NewJavascriptPackageCataloger"). Also, we might want to specify in the description that this cataloger looks at `package.json` files.

Reply:
Thanks, good catch.